c10/util/Half.h

```diff
@@ -322,12 +353,18 @@ inline uint16_t fp16_ieee_from_fp32_value(float f) {
   return static_cast<uint16_t>(
       (sign >> 16) |
       (shl1_w > UINT32_C(0xFF000000) ? UINT16_C(0x7E00) : nonsign));
+#endif
 }
 
 } // namespace detail
 
 struct alignas(2) Half {
-  unsigned short x;
+  union {
+#ifdef NATIVE_FP16
+    _Float16 y;
```

Comment:
Does it need to specify `-mfp16-format` to enable `_Float16`? See https://gcc.gnu.org/onlinedocs/gcc/Half-Precision.html